### DISCUSSION TITLE
Will 7 colors epaper be supported in S3 board? Such as  EC060KH3, EC078KH5?

### DISCUSSION THREAD

**Discussion by lanistor** [vroland/epdiy#399] [General]
It would be really cool if such a color screen could be driven.

![WechatIMG33](https://github.com/user-attachments/assets/5f8ce08f-2a0b-4c61-bbdf-e4c50d725f82)

This is achieved by means of color filters. The driving logic should be the same as the 16-grayscale black and white screen, but it requires a color algorithm converter. 
The FPC interface is also the same as the `060XCH`, and as far as I know, the rendering speed is the same as the black and white screen.

**Comment by martinberlin** [comment]
I made a humble approach for the Kaleido 6” display here
https://github.com/vroland/epdiy/tree/s3_color_implementation
you can read more details in the wiki
![40B4DE7F-6C53-4D31-A93D-ECC72DF4FD08](https://github.com/user-attachments/assets/3c08d3f0-02b6-4742-839e-d5b3a5c986c3)
![C6A2EC00-C69D-40B4-9C5B-F301C0C5046C](https://github.com/user-attachments/assets/9b7f0cc7-d12d-4508-ae84-e0e772783a83)

**Comment by lanistor** [comment]
@martinberlin This is really cool, I'll try it right away.

**Comment by martinberlin** [comment]
Great @lanistor  please give some feedback in the next weeks of your experience trying this.
I've made this branch mostly as a personal experimentation. Probably there are wiser ways to deal with color, interested to know how it was the experience for you.

**Comment by lanistor** [comment]
> Great @lanistor please give some feedback in the next weeks of your experience trying this. I've made this branch mostly as a personal experimentation. Probably there are wiser ways to deal with color, interested to know how it was the experience for you.

Thanks, i tested it for once, but didn't found a demo to rendering with 7 colors directly, i will test it few days later, for too much development work recently.

**Comment by martinberlin** [comment]
You can just try the www-image example.
This should download a JPG and call my clone of the función epd_draw_pixel then for every pixel will be aware of what filter is on top

**Comment by lanistor** [comment]
> You can just try the www-image example. This should download a JPG and call my clone of the función epd_draw_pixel then for every pixel will be aware of what filter is on top

Got it, thanks 👍

**Comment by martinberlin** [comment]
This has been way to long without any activity. You can find it converted to a discussion into General section
